core/templates/dev/head/services/ContextService.js

```diff
@@ -100,16 +117,44 @@ oppia.factory('ExplorationContextService', [
           }
 
           throw Error(
-            'ERROR: ExplorationContextService should not be used outside the ' +
-            'context of an exploration.');
+            'ERROR: ContextService should not be used outside the ' +
+            'context of an exploration or a question.');
+        }
+      },
+
+      // Returns a string representing the questionId (obtained from the
+      // URL).
+      getQuestionId: function() {
+        if (questionId) {
+          return questionId;
+        } else {
+          // The pathname should /question_editor/{question_id}.
+          var pathnameArray = UrlService.getPathname().split('/');
+          for (var i = 0; i < pathnameArray.length; i++) {
+            if (pathnameArray[i] === 'question_editor') {
+              questionId = pathnameArray[i + 1];
+              return pathnameArray[i + 1];
+            }
+          }
+
+          throw Error(
+            'ERROR: ContextService should not be used outside the ' +
+            'context of an exploration or a question.');
         }
       },
 
       // Following variable helps to know whether exploration editor is
       // in main editing mode or preview mode.
       isInExplorationEditorMode: function() {
-        return (this.getPageContext() === PAGE_CONTEXT.EDITOR &&
-            this.getEditorTabContext() === EDITOR_TAB_CONTEXT.EDITOR);
+        return (this.getPageContext() === PAGE_CONTEXT.EXPLORATION_EDITOR &&
+            this.getEditorTabContext() === (
+              EDITOR_TAB_CONTEXT.EXPLORATION_EDITOR));
+      },
+
+      // Following variable helps to know whether question editor is
```

Comment:
??

Reply:
I think we can skip having this method, as there's only editing mode in question editor for now. It can be added if needed later.